src/main/java/HttpRequest.java

```diff
@@ -0,0 +1,38 @@
+package com.td.HttpServer;
+
+public class HttpRequest implements HttpProtocal {
+
+  private String method;
+  private String path;
+  private String version;
+
+  public HttpRequest(String request) {
+    String request_array[] = request.split("\\s+");
+    method = request_array[0];
+    path = request_array[1];
+    version = request_array[2];
+
+    // TODO:: add logic to store Request Header Fields and optional message body
```

Comment:
I am not fully sure about all the possible additional Fields that can be passed sent during an HTTP request. The todo is here as a reminded that I will probably have to add additional features to deal with this in the future.